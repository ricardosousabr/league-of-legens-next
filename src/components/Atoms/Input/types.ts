import React, { ReactNode } from 'react';

export interface InputProps extends IntrinsicElements.input DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {
  type: string,
  value: string,
  placeholder: string,
  onChange: function,
}