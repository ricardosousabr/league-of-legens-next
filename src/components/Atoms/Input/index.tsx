import { InputProps } from "./types"

export default function Input({ type, placeholder, value, disabled }:InputProps) {
  return <input disabled={disabled} value={value} type={type} placeholder={placeholder} />
}