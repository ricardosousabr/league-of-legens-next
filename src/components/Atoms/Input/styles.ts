import styled from "styled-components";
import { InputProps } from "./types"

export const InputStyle = styled.input<InputProps>`
  border-radius: ${(p) => p.borderRadius};
  background: ${(p) => p.background} ;
  padding: ${(p) => p.padding};
  border: ${(p) => p.border};
  border-image: linear-gradient(45deg, #ffc600, #bf930d) 1;

  &:focus {
    border: ${(p) => p.border};
    border-image: linear-gradient(45deg, #ffc600, #bf930d) 1;
    outline: 0;
  }
`