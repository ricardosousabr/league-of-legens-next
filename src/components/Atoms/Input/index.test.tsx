import renderer from "react-test-renderer";
import Input from "./index";

test("Link renders correctly", () => {
  const tree = renderer.create(<Input type="submit" placeholder="Click aqui" value="Click" />).toJSON();
  expect(tree).toMatchSnapshot();
});
