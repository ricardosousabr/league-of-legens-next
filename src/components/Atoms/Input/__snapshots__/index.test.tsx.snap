// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<input
  className="styles__InputStyle-sc-pk8aov-0 feybKy"
  placeholder="Click aqui"
  type="submit"
  value="Click"
/>
`;
